commit a2e31d0681d8a47389b8a3552622fbd9827bcef4
Author: ktkachov <ktkachov@138bc75d-0d04-0410-961f-82ee72b054a4>
Date:   Wed Apr 22 14:20:01 2015 +0000

    unwind fix for musl
    
    On behalf of szabolcs.nagy@arm.com
    
    2015-04-22  Gregor Richards  <gregor.richards@uwaterloo.ca>
    	    Szabolcs Nagy  <szabolcs.nagy@arm.com>
    
    	* unwind-dw2-fde-dip.c (USE_PT_GNU_EH_FRAME): Define it on
    	Linux if target provides dl_iterate_phdr.
    
    
    git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@222328 138bc75d-0d04-0410-961f-82ee72b054a4

diff --git a/libgcc/unwind-dw2-fde-dip.c b/libgcc/unwind-dw2-fde-dip.c
index e1e566b..137dced 100644
--- a/libgcc/unwind-dw2-fde-dip.c
+++ b/libgcc/unwind-dw2-fde-dip.c
@@ -59,6 +59,12 @@
 
 #if !defined(inhibit_libc) && defined(HAVE_LD_EH_FRAME_HDR) \
     && defined(TARGET_DL_ITERATE_PHDR) \
+    && defined(__linux__)
+# define USE_PT_GNU_EH_FRAME
+#endif
+
+#if !defined(inhibit_libc) && defined(HAVE_LD_EH_FRAME_HDR) \
+    && defined(TARGET_DL_ITERATE_PHDR) \
     && (defined(__DragonFly__) || defined(__FreeBSD__))
 # define ElfW __ElfN
 # define USE_PT_GNU_EH_FRAME
